File: DesignPatternsExercises/ruby/CommandDesignPattern/tsv_extraction.rb
require_relative 'extraction'
require_relative 'tsv_extractor'

class TSVExtraction < Extraction

  attr_accessor :file

  def initialize(file)
    @file = file
  end

  def call_correct_extraction
    # __________(8)_____________
    TSVExtractor.new.extract
  end

end

File: DesignPatternsExercises/java/CommandDesignPattern/Solution/TsvExtraction.java
// class TsvExtraction implements __________(9)__________ {
class TsvExtraction implements Extraction {
	Extractor tsvExtractor = new TsvExtractor();

	@Override
	public void callCorrectExtraction() {
		// tsvExtractor.__________(10)__________();
		tsvExtractor.extract();
	}
}
